src/sqlancer/databend/ast/DatabendConstant.java

```diff
@@ -318,7 +322,10 @@ public static class DatabendDateConstant extends DatabendConstant {
         public String textRepr;
 
         public DatabendDateConstant(long val) {
-            Timestamp timestamp = new Timestamp(val);
+            // Databend supports `date` type where the year cannot exceed `9999`,
+            // the value is truncated to ensure generate legitimate `date` value.
+            long t = val % 253380000000000L;
```

Comment:
Thanks, for merging this after the tests pass. As a minor nitpick: I think a named constant would have been better as we have the same comment and truncation factor twice in the code.